components/automate-ui/src/app/modules/token/token-list/api-token-list.component.ts

```diff
@@ -132,22 +133,20 @@ export class ApiTokenListComponent implements OnInit, OnDestroy {
   public createToken(): void {
     this.creatingToken = true;
     const tok = {
-      id: this.createTokenForm.controls['id'].value,
-      name: this.createTokenForm.controls['name'].value.trim(),
+      id: this.createTokenForm.controls.id.value,
+      name: this.createTokenForm.controls.name.value.trim(),
       projects: this.createTokenForm.controls.projects.value
     };
     this.store.dispatch(new CreateToken(tok));
+  }
 
-    const member = stringToMember(`token:${tok.id}`);
-    const policies: string[] = this.createTokenForm.controls.policies.value;
-    if (policies) {
-      policies.forEach(id =>
-        this.store.dispatch(new AddPolicyMembers(<PolicyMembersMgmtPayload>{
-          id,
-          members: [member]
-        }))
-      );
-    }
+  private assignPolicies(): void {
+    const member = stringToMember(`token:${this.createTokenForm.controls.id.value}`);
+    const policies: string[] = this.createTokenForm.controls.policies.value || [];
+    policies.forEach(id => this.store.dispatch(new AddPolicyMembers(<PolicyMembersMgmtPayload>{
```

Comment:
❔ What does the cast give us here? (Sorry, I can see it's moved code, just wondering.)

Reply:
Actually, it is unneeded here--good eye!